### DISCUSSION TITLE
重构 ModelProviders 面向对象设计

### DISCUSSION THREAD

**Discussion by BrandonStudio** [lobehub/lobehub#4634] [RFCs]
- 重新设计 ModelProviders 封装模式，令非公有字段（如 `client`、`baseURL`）变为 `protected`，把可重写的私有字段/方法也变为 `protected`
- 重构 `LobeOpenAICompatibleAI`，将工厂函数类生成转换为抽象父类派生。导出 `LobeOpenAICompatibleAI` 并为其添加泛型

**Comment by arvinxx** [comment]
我感觉继承类的方案不好，还是 createOpenAI('xAI') 这样的设计比较理想
